lib/u3d/unity_module.rb

```diff
@@ -0,0 +1,132 @@
+## --- BEGIN LICENSE BLOCK ---
+# Copyright (c) 2016-present WeWantToKnow AS
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+## --- END LICENSE BLOCK ---
+
+module U3d
+  class UnityModule
+    # Basic module attributes
+    attr_reader :id, :name, :description, :url
+    # Validation attributes
+    attr_reader :installed_size, :download_size, :checksum
+
+    def initialize(
+      id:, name:, description:, url:,
+      installed_size:, download_size:, checksum:
+    )
+      @id = id
+      @name = name
+      @description = description
+      @url = url
+      @installed_size = installed_size
+      @download_size = download_size
+      @checksum = checksum
+    end
+
+    def download_size_bytes(_os)
+      size_in_bytes(download_size)
+    end
+
+    def installed_size_bytes(_os)
+      size_in_bytes(installed_size)
+    end
+
+    class << self
+      def load_modules(version, cached_versions, os: U3dCore::Helper.operating_system, offline: false)
+        if version.is_a? Array
+          UI.verbose "Loading modules for several versions: #{version}"
+          load_versions_modules(version, cached_versions, os, offline)
+        else
+          UI.verbose "Loading modules for version #{version}"
```

Comment:
It would change the interface though.

`load_version_module` returns an array of modules, while `load_versions_module` returns an hash of version, array of modules.
Maybe we want to clean that up though.